app-engine/java/src/com/google/cloud/tools/intellij/appengine/java/util/AppEngineUtil.java

```diff
@@ -147,6 +151,34 @@ private static boolean doesArtifactMatchEnvironment(
     return moduleDeploymentSources;
   }
 
+  /**
+   * Assembles a list of {@link GradlePluginDeploymentSource} for each module.
+   *
+   * <p>Will create a Gradle plugin deployment source if the module has both the App Engine standard
+   * the App Engine Gradle plugin facet.
+   *
+   * @param project the current {@link Project}
+   * @return a list of {@link ModuleDeploymentSource} containing the app-gradle-plugin based
+   *     deployment sources.
+   */
+  public static List<ModuleDeploymentSource> createGradlePluginDeploymentSources(
+      @NotNull Project project) {
+    List<ModuleDeploymentSource> moduleDeploymentSources = Lists.newArrayList();
+
+    Stream.of(ModuleManager.getInstance(project).getModules())
+        .forEach(
+            module -> {
+              if (AppEngineStandardFacet.hasFacet(module)
```

Comment:
could you please remind me how these two facets are cooperating - both of them are supposed to be present in the module for deployment, but App Engine Standard is also available in Community regardless of Web support presence? For some reason I thought opposite.

Reply:
The `App Engine standard facet` is there always - if its a standard project (appengine-web.xml) - regardless of CE or UE.

The `App Engine gradle plugin facet` is there only in Community Edition _if_ the module has the app-gradle-plugin AND the `App Engine standard facet` is there too.

For deployment from this new gradle source, _both_ facets are required.

In reality, the `App Engine gradle plugin facet` is a dependent facet, or a "child" of the app engine facet. There is the concept of an "underlying" facet - I'll see if this can better represent this relationship.

Reply:
yeah it seems that using an underlying facet will better convey and enforce this relationship. let me refactor this.

Reply:
nevermind, it seems that you can only initialize the underlying facet at facet creation time, meaning that we would need to initialize the gradle facet when the app engine standard facet is created. This is not what we want because the gradle facet is tied to the gradle sync.